### DISCUSSION TITLE
Issue with UDP throughput and data loss in iperf3 tests

### DISCUSSION THREAD

**Discussion by adilnaveed** [esnet/iperf#1722] [Q&A]
Hello,

I am encountering an issue while conducting UDP throughput tests using iperf3 between a client and server setup. Here are the details of the problem:

Test Configuration:

Client: Amarisoft UE simulator acting as multiple clients, sending UDP data to the server.
Server: Receiving UDP traffic from multiple clients on different ports.
Test Parameters: Bitrate set to 1M and UDP buffer size (-w) configured as 2M.
Observations:

UDP Test (Client to Server):
Approximately half of the data sent from clients to the server is lost.
The server does not receive the expected 1M bitrate; instead, there is significant data loss.
UDP Test (Server to Client, using -R):
Data loss is not observed when testing in the reverse direction (server to clients).
TCP Tests:
No data loss is observed in either direction (client to server or server to client).
Question:

Issue Context: The problem appears to be specific to UDP tests from clients to the server with the chosen bitrate and buffer size configuration.
Hypothesis: What could be causing the data loss when using UDP in one direction while the reverse direction and TCP tests show no data loss?
Could you please provide insights or recommendations on how to mitigate this issue? Your assistance would be greatly appreciated.

**Comment by davidBar-On** [comment]
Hi, just to make sure, can you add some information:
1. Can you show the iperf3 commands you are using (no need to show the specific IP addresses)?
2. As the simulator is "acting as multiple clients", how many clients are active in parallel?  What is the the bitrate (`-b`) set for each?
3. What cellular network is used (LTE/4G? 5G)?

From what I read, the Amarisoft UE simulator works over cellular network.  In these networks, the downlink throughput (server to client) is much higher than the uplink (client to server) throughput.  Can it be that uplink does not support the required 1Mbps, maybe because the uplink is busy because of other data sent over it?

If this is not the case, more analysis is required:
1. Does Amarisoft  support logging that may help with such analysis?
2. What happens if you change the sent bitrate from the client to 256K bits (`-b 256K`)?  512Kbits?
3. What happens if you increase/decrease the packet length (using `-l` - the default is usually 1460 bytes)?
4. Can you use tools like Wireshark to analyze the traffic?  It can help to see which packets are lost, are they lost in blocks or one arrive and one lost, etc.

**Comment by xia-s-xian** [comment]
Hi,

I have also encountered the same problem, when using UDP upstream test, the packet loss rate reached 50%, but the TCP upstream and downstream, UDP downstream are normal.

At present, I find that it is caused by the problem of the window environment. It is normal for me to switch to the ubuntu computer as the server test, but I do not know what causes the problem, and there is no effective way to confirm it at present.

Testing environment:
EVB Board(usb3.0)->USB to network adapter->window PC(ETH)
EVB Board(client),window PC(serve)
![image](https://github.com/user-attachments/assets/6191da25-ed7c-4a02-9ba8-515007467bcd)

EVB Board(usb3.0)->USB to network adapter->Ubuntu PC(ETH)
EVB Board(client),Ubuntu  PC(serve)
![image](https://github.com/user-attachments/assets/180ff064-1cb8-4d7a-b86f-6c8d90fe1b3a)

I tried to change the iperf version on the PC, but nothing worked. And the firewalls are down.

I'd like to know the root cause of this problem.

**Comment by davidBar-On** [comment]
Hi @xia-s-xian, several reasons can cause this issue, so more information is needed:

1. Run the clients and servers using -V option.  What is block size used (the " byte blocks" size)?  What is the iperf3 versions used?
2. Run TCP tests (in both environments) setting `-l` to the "bytes blocks" size used by the UDP tests and without limiting the throughput (i.e. without setting the `-b` option.  What is the throughput (in both cases)?   That throughput is the approximate maximum possible throughput.
3. The sending (client) side throughput for UDP is only limited by the sending computer resources (CPU, buffers, etc.).  Therefore it is strange that the sending rate in your first test is different then the second test.  Did you use the same EVB and USB adapter in both tests?
4.  As you wrote, it may be that the problem is in the Windows PC.  May be it is slower, ha limited resources, etc.  It may also be because iperf3 under Windows is running over Cygwin.  However, I assume that Ubuntu is using a much newer version of iperf3 and it is recommended to use a much newer iperf3 version before running the above tests.

**Comment by xia-s-xian** [comment]
Hi,@davidBar-On:

The situation I tested updating to iperf3.17.1 is as follows（using -V option and tcp tests -l to the “bytes  blocks”）:

- - - - - - - - - - - - - - - - - - - - - - - - -

**EVB Board(usb3.0)->USB to network adapter->Window PC(ETH)
UDP TEST：**

root@*:~#./iperf3 -V -c 169.254.146.6 -u -b 1000M -t 10
iperf 3.17.1
Linux * 5.10.59 #1 SMP Fri Jun 28 10:50:09 CST 2024 aarch64
Control connection MSS 1448
Setting UDP block size to 1448
Time: Sat, 13 Jul 2024 10:52:51 GMT
Connecting to host 169.254.146.6, port 5201
      Cookie: cfumrj575hlnmh2ira4kmrks57pvu4n73a55
      Target Bitrate: 1000000000
[  5] local 169.254.10.10 port 41187 connected to 169.254.146.6 port 5201
Starting Test: protocol: UDP, 1 streams, 1448 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate         Total Datagrams
[  5]   0.00-1.00   sec   119 MBytes   999 Mbits/sec  86270
[  5]   1.00-2.00   sec   107 MBytes   899 Mbits/sec  77646
[  5]   2.00-3.00   sec   107 MBytes   900 Mbits/sec  77651
[  5]   3.00-4.00   sec   124 MBytes  1.04 Gbits/sec  90002
[  5]   4.00-5.00   sec   124 MBytes  1.04 Gbits/sec  89858
[  5]   5.00-6.00   sec  83.0 MBytes   696 Mbits/sec  60081
[  5]   6.00-7.00   sec  71.9 MBytes   603 Mbits/sec  52093
[  5]   7.00-8.00   sec   106 MBytes   893 Mbits/sec  77121
[  5]   8.00-9.00   sec   125 MBytes  1.05 Gbits/sec  90337
[  5]   9.00-10.00  sec   123 MBytes  1.03 Gbits/sec  89173
Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5]   0.00-10.00  sec  1.07 GBytes   915 Mbits/sec  0.000 ms  0/790232 (0%)  sender
[  5]   0.00-10.00  sec  8.58 MBytes  7.20 Mbits/sec  0.008 ms  7077/13289 (53%)  receiver
CPU Utilization: local/sender 79.1% (7.8%u/71.3%s), remote/receiver 11.2% (3.7%u/7.5%s)

iperf Done.

iperf 3.17.1
CYGWIN_NT-10.0-22631 test0018 3.5.3-1.x86_64 2024-04-03 17:25 UTC x86_64
Server listening on 5201 (test #5)
Time: Sat, 13 Jul 2024 10:52:21 GMT
Accepted connection from 169.254.10.10, port 50596
      Cookie: cfumrj575hlnmh2ira4kmrks57pvu4n73a55
      Target Bitrate: 1000000000
[  5] local 169.254.146.6 port 5201 connected to 169.254.10.10 port 41187
Starting Test: protocol: UDP, 1 streams, 1448 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5]   0.00-1.00   sec  8.58 MBytes  71.9 Mbits/sec  0.008 ms  7077/13289 (53%)
[  5]   1.00-2.00   sec  0.00 Bytes  0.00 bits/sec  0.008 ms  0/0 (0%)
[  5]   2.00-3.00   sec  0.00 Bytes  0.00 bits/sec  0.008 ms  0/0 (0%)
[  5]   3.00-4.00   sec  0.00 Bytes  0.00 bits/sec  0.008 ms  0/0 (0%)
[  5]   4.00-5.00   sec  0.00 Bytes  0.00 bits/sec  0.008 ms  0/0 (0%)
[  5]   5.00-6.00   sec  0.00 Bytes  0.00 bits/sec  0.008 ms  0/0 (0%)
[  5]   6.00-7.00   sec  0.00 Bytes  0.00 bits/sec  0.008 ms  0/0 (0%)
[  5]   7.00-8.00   sec  0.00 Bytes  0.00 bits/sec  0.008 ms  0/0 (0%)
[  5]   8.00-9.00   sec  0.00 Bytes  0.00 bits/sec  0.008 ms  0/0 (0%)
[  5]   9.00-10.00  sec  0.00 Bytes  0.00 bits/sec  0.008 ms  0/0 (0%)
Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5] (sender statistics not available)
[  5]   0.00-10.00  sec  8.58 MBytes  7.20 Mbits/sec  0.008 ms  7077/13289 (53%)  receiver
iperf 3.17.1

- - - - - - - - - - - - - - - - - - - - - - - - -

**TCP TEST：**

root@*:~#./iperf3 -V -c 169.254.146.6 -l 1448  -t 10
iperf 3.17.1
Linux * 5.10.59 #1 SMP Fri Jun 28 10:50:09 CST 2024 aarch64
Control connection MSS 1448
Time: Sat, 13 Jul 2024 11:03:21 GMT
Connecting to host 169.254.146.6, port 5201
      Cookie: ke32iwcc7rkqjond2zfl76jmbwdaeauq4fyt
      TCP MSS: 1448 (default)
[  5] local 169.254.10.10 port 50610 connected to 169.254.146.6 port 5201
Starting Test: protocol: TCP, 1 streams, 1448 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec  59.5 MBytes   499 Mbits/sec    0   1.11 MBytes
[  5]   1.00-2.00   sec  55.1 MBytes   462 Mbits/sec    0   1.11 MBytes
[  5]   2.00-3.00   sec  49.5 MBytes   415 Mbits/sec    0   1.11 MBytes
[  5]   3.00-4.00   sec  52.9 MBytes   444 Mbits/sec    0   1.11 MBytes
[  5]   4.00-5.00   sec  49.5 MBytes   415 Mbits/sec    0   1.11 MBytes
[  5]   5.00-6.00   sec  48.8 MBytes   410 Mbits/sec    0   1.11 MBytes
[  5]   6.00-7.00   sec  50.2 MBytes   421 Mbits/sec    0   1.11 MBytes
[  5]   7.00-8.00   sec  46.7 MBytes   392 Mbits/sec    0   1.11 MBytes
[  5]   8.00-9.00   sec  45.2 MBytes   379 Mbits/sec    0   1.11 MBytes
[  5]   9.00-10.00  sec  51.1 MBytes   429 Mbits/sec    0   1.11 MBytes

Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec   509 MBytes   427 Mbits/sec    0             sender
[  5]   0.00-10.01  sec   504 MBytes   423 Mbits/sec                  receiver
CPU Utilization: local/sender 9.5% (1.9%u/7.6%s), remote/receiver 12.0% (3.0%u/9.0%s)
snd_tcp_congestion cubic

iperf Done.

iperf 3.17.1
CYGWIN_NT-10.0-22631 test0018 3.5.3-1.x86_64 2024-04-03 17:25 UTC x86_64
Server listening on 5201 (test #1)
Time: Sat, 13 Jul 2024 11:02:51 GMT
Accepted connection from 169.254.10.10, port 50608
      Cookie: ke32iwcc7rkqjond2zfl76jmbwdaeauq4fyt
      TCP MSS: 0 (default)
[  5] local 169.254.146.6 port 5201 connected to 169.254.10.10 port 50610
Starting Test: protocol: TCP, 1 streams, 1448 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-1.01   sec  54.6 MBytes   452 Mbits/sec
[  5]   1.01-2.01   sec  54.7 MBytes   462 Mbits/sec
[  5]   2.01-3.02   sec  50.4 MBytes   420 Mbits/sec
[  5]   3.02-4.01   sec  52.5 MBytes   442 Mbits/sec
[  5]   4.01-5.01   sec  49.8 MBytes   420 Mbits/sec
[  5]   5.01-6.00   sec  47.6 MBytes   401 Mbits/sec
[  5]   6.00-7.01   sec  49.9 MBytes   416 Mbits/sec
[  5]   7.01-8.01   sec  47.8 MBytes   398 Mbits/sec
[  5]   8.01-9.01   sec  44.6 MBytes   377 Mbits/sec
[  5]   9.01-10.01  sec  52.6 MBytes   439 Mbits/sec
Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate
[  5] (sender statistics not available)
[  5]   0.00-10.01  sec   504 MBytes   423 Mbits/sec                  receiver
iperf 3.17.1


- - - - - - - - - - - - - - - - - - - - - - - - -

**EVB Board(usb3.0)->USB to network adapter->Ubuntu PC(ETH)
UDP TEST：**

root@*:~#./iperf3 -V -c 169.254.146.3 -u -b 1000M -t 10
iperf 3.17.1
Linux * 5.10.59 #1 SMP Fri Jun 28 10:50:09 CST 2024 aarch64
Control connection MSS 1448
Setting UDP block size to 1448
Time: Sat, 13 Jul 2024 10:59:32 GMT
Connecting to host 169.254.146.3, port 5201
      Cookie: ut74lzc77acxngkw4ox2eyquaivbsozkjynb
      Target Bitrate: 1000000000
[  5] local 169.254.10.10 port 45340 connected to 169.254.146.3 port 5201
Starting Test: protocol: UDP, 1 streams, 1448 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate         Total Datagrams
[  5]   0.00-1.00   sec   119 MBytes  1000 Mbits/sec  86314
[  5]   1.00-2.00   sec   119 MBytes  1.00 Gbits/sec  86341
[  5]   2.00-3.00   sec   119 MBytes  1000 Mbits/sec  86312
[  5]   3.00-4.00   sec   119 MBytes  1.00 Gbits/sec  86342
[  5]   4.00-5.00   sec   119 MBytes   999 Mbits/sec  86269
[  5]   5.00-6.00   sec   119 MBytes  1000 Mbits/sec  86321
[  5]   6.00-7.00   sec   119 MBytes  1.00 Gbits/sec  86374
[  5]   7.00-8.00   sec   119 MBytes  1000 Mbits/sec  86310
[  5]   8.00-9.00   sec   119 MBytes  1.00 Gbits/sec  86329
[  5]   9.00-10.00  sec   119 MBytes   999 Mbits/sec  86288
Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5]   0.00-10.00  sec  1.16 GBytes  1000 Mbits/sec  0.000 ms  0/863200 (0%)  sender
[  5]   0.00-10.00  sec  1.16 GBytes  1000 Mbits/sec  0.014 ms  0/863200 (0%)  receiver
CPU Utilization: local/sender 100.4% (15.2%u/85.2%s), remote/receiver 31.6% (4.3%u/27.2%s)

iperf Done.

iperf 3.17.1
Linux xsx-Inspiron-5509 5.15.0-113-generic  123~20.04.1-Ubuntu SMP Wed Jun 12 17:33:13 UTC 2024 x86_64
Server listening on 5201 (test #1)
Time: Sat, 13 Jul 2024 10:59:32 GMT
Accepted connection from 169.254.10.10, port 50412
      Cookie: ut74lzc77acxngkw4ox2eyquaivbsozkjynb
      Target Bitrate: 1000000000
[  5] local 169.254.146.3 port 5201 connected to 169.254.10.10 port 45340
Starting Test: protocol: UDP, 1 streams, 1448 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5]   0.00-1.00   sec   119 MBytes   999 Mbits/sec  0.013 ms  0/86282 (0%)
[  5]   1.00-2.00   sec   119 MBytes  1.00 Gbits/sec  0.014 ms  0/86340 (0%)
[  5]   2.00-3.00   sec   119 MBytes  1000 Mbits/sec  0.015 ms  0/86312 (0%)
[  5]   3.00-4.00   sec   119 MBytes  1.00 Gbits/sec  0.014 ms  0/86344 (0%)
[  5]   4.00-5.00   sec   119 MBytes   999 Mbits/sec  0.014 ms  0/86270 (0%)
[  5]   5.00-6.00   sec   119 MBytes  1000 Mbits/sec  0.014 ms  0/86324 (0%)
[  5]   6.00-7.00   sec   119 MBytes  1.00 Gbits/sec  0.014 ms  0/86375 (0%)
[  5]   7.00-8.00   sec   119 MBytes  1000 Mbits/sec  0.014 ms  0/86309 (0%)
[  5]   8.00-9.00   sec   119 MBytes  1.00 Gbits/sec  0.014 ms  0/86331 (0%)
[  5]   9.00-10.00  sec   119 MBytes  1000 Mbits/sec  0.013 ms  0/86291 (0%)
[  5]  10.00-10.00  sec  29.7 KBytes   429 Mbits/sec  0.014 ms  0/22 (0%)
Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5] (sender statistics not available)
[  5]   0.00-10.00  sec  1.16 GBytes  1000 Mbits/sec  0.014 ms  0/863200 (0%)  receiver
iperf 3.17.1

- - - - - - - - - - - - - - - - - - - - - - - - -

**TCP TEST：**

root@*:~#./iperf3 -V -c 169.254.146.3 -l 1448  -t 10
iperf 3.17.1
Linux * 5.10.59 #1 SMP Fri Jun 28 10:50:09 CST 2024 aarch64
Control connection MSS 1448
Time: Sat, 13 Jul 2024 11:00:49 GMT
Connecting to host 169.254.146.3, port 5201
      Cookie: 7aleevzcq5lga2bf456clrppwuy6t2ra5kfa
      TCP MSS: 1448 (default)
[  5] local 169.254.10.10 port 50416 connected to 169.254.146.3 port 5201
Starting Test: protocol: TCP, 1 streams, 1448 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec  86.5 MBytes   725 Mbits/sec    0   33.9 KBytes
[  5]   1.00-2.00   sec  89.5 MBytes   751 Mbits/sec    0   33.9 KBytes
[  5]   2.00-3.00   sec  89.6 MBytes   751 Mbits/sec    0   33.9 KBytes
[  5]   3.00-4.00   sec  89.5 MBytes   751 Mbits/sec    0   33.9 KBytes
[  5]   4.00-5.00   sec  89.7 MBytes   752 Mbits/sec    0   33.9 KBytes
[  5]   5.00-6.00   sec  90.3 MBytes   757 Mbits/sec    0   33.9 KBytes
[  5]   6.00-7.00   sec  89.8 MBytes   754 Mbits/sec    0   45.2 KBytes
[  5]   7.00-8.00   sec  90.0 MBytes   755 Mbits/sec    0   45.2 KBytes
[  5]   8.00-9.00   sec  90.0 MBytes   755 Mbits/sec    0   45.2 KBytes
[  5]   9.00-10.00  sec  90.0 MBytes   755 Mbits/sec    0   45.2 KBytes
Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec   895 MBytes   751 Mbits/sec    0             sender
[  5]   0.00-10.00  sec   895 MBytes   750 Mbits/sec                  receiver
CPU Utilization: local/sender 99.5% (4.6%u/94.9%s), remote/receiver 39.7% (2.0%u/37.7%s)
snd_tcp_congestion cubic
rcv_tcp_congestion cubic

iperf Done.

iperf 3.17.1
Linux xsx-Inspiron-5509 5.15.0-113-generic #123~20.04.1-Ubuntu SMP Wed Jun 12 17:33:13 UTC 2024 x86_64
Server listening on 5201 (test #2)
Time: Sat, 13 Jul 2024 11:00:49 GMT
Accepted connection from 169.254.10.10, port 50414
      Cookie: 7aleevzcq5lga2bf456clrppwuy6t2ra5kfa
      TCP MSS: 0 (default)
[  5] local 169.254.146.3 port 5201 connected to 169.254.10.10 port 50416
Starting Test: protocol: TCP, 1 streams, 1448 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-1.00   sec  86.4 MBytes   724 Mbits/sec
[  5]   1.00-2.00   sec  89.5 MBytes   751 Mbits/sec
[  5]   2.00-3.00   sec  89.6 MBytes   752 Mbits/sec
[  5]   3.00-4.00   sec  89.5 MBytes   751 Mbits/sec
[  5]   4.00-5.00   sec  89.7 MBytes   752 Mbits/sec
[  5]   5.00-6.00   sec  90.3 MBytes   757 Mbits/sec
[  5]   6.00-7.00   sec  89.8 MBytes   754 Mbits/sec
[  5]   7.00-8.00   sec  90.0 MBytes   755 Mbits/sec
[  5]   8.00-9.00   sec  90.0 MBytes   755 Mbits/sec
[  5]   9.00-10.00  sec  90.0 MBytes   755 Mbits/sec
[  5]  10.00-10.00  sec  38.2 KBytes   106 Mbits/sec
Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate
[  5] (sender statistics not available)
[  5]   0.00-10.00  sec   895 MBytes   750 Mbits/sec                  receiver
rcv_tcp_congestion cubic


Did you use the same EVB and USB adapter in both tests?
->Yes.

**Comment by davidBar-On** [comment]
Hi, from the TCP tests it seems that the throughput limitation is in the Windows PC - either the in Windows or in iperf3.  **Are you able to try the TCP tests using iperf (iperf2)?**  iperf is not using Cygwin, so that may show whether the problem is in iperf3 or Windows (or the Windows PC).

Regarding the UDP test using Windows.  It seems that the server is stuck.  This is either iperf3 or Cygwin problem (note that iperf3 is not supported formally under Windows).  **Can you try running few UDP tests using Windows with limiting the client bit rate to 300M, 500M, 700M?**  This may show whether this is an iperf3 problem in all bit rates.

**Comment by xia-s-xian** [comment]
Hi, David:

1.It is normal to test udp using iperf2 version. It should be a problem with iperf3 version.

:~$ iperf -c 169.254.254.188 -i 1  -u -b 1000M
Client connecting to 169.254.254.188, UDP port 5001
Sending 1470 byte datagrams, IPG target: 11.22 us (kalman adjust)
UDP buffer size:  208 KByte (default)
[  3] local 169.254.254.180 port 57597 connected with 169.254.254.188 port 5001
[ ID] Interval       Transfer     Bandwidth
[  3]  0.0- 1.0 sec   113 MBytes   944 Mbits/sec
[  3]  1.0- 2.0 sec   112 MBytes   942 Mbits/sec
[  3]  2.0- 3.0 sec   113 MBytes   946 Mbits/sec
[  3]  3.0- 4.0 sec   113 MBytes   950 Mbits/sec
[  3]  4.0- 5.0 sec   113 MBytes   949 Mbits/sec
[  3]  5.0- 6.0 sec   112 MBytes   941 Mbits/sec
[  3]  6.0- 7.0 sec   113 MBytes   949 Mbits/sec
[  3]  7.0- 8.0 sec   112 MBytes   938 Mbits/sec
[  3]  8.0- 9.0 sec   112 MBytes   941 Mbits/sec
[  3]  9.0-10.0 sec   113 MBytes   951 Mbits/sec
[  3]  0.0-10.0 sec  1.10 GBytes   945 Mbits/sec
[  3] Sent 803639 datagrams
[  3] Server Report:
[  3]  0.0-10.0 sec  1.10 GBytes   945 Mbits/sec   0.021 ms  280/803639 (0.035%)

 .\iperf-2.2.n-win64.exe -s -u -i 1
Server listening on UDP port 5001
UDP buffer size: 64.0 KByte (default)
[  1] local 169.254.254.188 port 5001 connected with 169.254.254.180 port 57597
[ ID] Interval       Transfer     Bandwidth        Jitter   Lost/Total Datagrams
[  1] 0.00-1.00 sec   112 MBytes   942 Mbits/sec   0.017 ms 146/80287 (0.18%)
[  1] 1.00-2.00 sec   112 MBytes   942 Mbits/sec   0.011 ms 0/80094 (0%)
[  1] 2.00-3.00 sec   113 MBytes   946 Mbits/sec   0.011 ms 0/80427 (0%)
[  1] 3.00-4.00 sec   113 MBytes   950 Mbits/sec   0.008 ms 0/80795 (0%)
[  1] 4.00-5.00 sec   113 MBytes   950 Mbits/sec   0.020 ms 0/80749 (0%)
[  1] 5.00-6.00 sec   112 MBytes   941 Mbits/sec   0.012 ms 4/80039 (0.005%)
[  1] 6.00-7.00 sec   113 MBytes   949 Mbits/sec   0.014 ms 8/80708 (0.0099%)
[  1] 7.00-8.00 sec   112 MBytes   937 Mbits/sec   0.030 ms 122/79762 (0.15%)
[  1] 8.00-9.00 sec   112 MBytes   940 Mbits/sec   0.013 ms 0/79965 (0%)
[  1] 9.00-10.00 sec   113 MBytes   951 Mbits/sec   0.018 ms 0/80813 (0%)
[  1] 0.00-10.00 sec  1.10 GBytes   945 Mbits/sec   0.022 ms 280/803639 (0.035%)


2.Try running few UDP tests using Windows with limiting the client bit rate to 300M, 500M, 700M.

ubuntu:~$  iperf3  -c 169.254.254.188 -u -b 300M -t 10
Connecting to host 169.254.254.188, port 5201
[  5] local 169.254.254.180 port 39559 connected to 169.254.254.188 port 5201
[ ID] Interval           Transfer     Bitrate         Total Datagrams
[  5]   0.00-1.00   sec  35.7 MBytes   300 Mbits/sec  25884
[  5]   1.00-2.00   sec  35.8 MBytes   300 Mbits/sec  25898
[  5]   2.00-3.00   sec  35.8 MBytes   300 Mbits/sec  25898
[  5]   3.00-4.00   sec  35.8 MBytes   300 Mbits/sec  25896
[  5]   4.00-5.00   sec  35.8 MBytes   300 Mbits/sec  25900
[  5]   5.00-6.00   sec  35.8 MBytes   300 Mbits/sec  25897
[  5]   6.00-7.00   sec  35.8 MBytes   300 Mbits/sec  25896
[  5]   7.00-8.00   sec  35.8 MBytes   300 Mbits/sec  25900
[  5]   8.00-9.00   sec  35.8 MBytes   300 Mbits/sec  25898
[  5]   9.00-10.00  sec  35.8 MBytes   300 Mbits/sec  25892
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5]   0.00-10.00  sec   358 MBytes   300 Mbits/sec  0.000 ms  0/258959 (0%)  sender
[  5]   0.00-10.04  sec   346 MBytes   289 Mbits/sec  0.022 ms  8585/258959 (3.3%)  receiver
iperf Done.

ubuntu:~$ iperf3  -c 169.254.254.188 -u -b 500M -t 10
Connecting to host 169.254.254.188, port 5201
[  5] local 169.254.254.180 port 42525 connected to 169.254.254.188 port 5201
[ ID] Interval           Transfer     Bitrate         Total Datagrams
[  5]   0.00-1.00   sec  59.6 MBytes   500 Mbits/sec  43160
[  5]   1.00-2.00   sec  59.6 MBytes   500 Mbits/sec  43166
[  5]   2.00-3.00   sec  59.6 MBytes   500 Mbits/sec  43150
[  5]   3.00-4.00   sec  59.6 MBytes   500 Mbits/sec  43174
[  5]   4.00-5.00   sec  59.6 MBytes   500 Mbits/sec  43152
[  5]   5.00-6.00   sec  59.6 MBytes   500 Mbits/sec  43155
[  5]   6.00-7.00   sec  59.6 MBytes   500 Mbits/sec  43169
[  5]   7.00-8.00   sec  59.6 MBytes   500 Mbits/sec  43177
[  5]   8.00-9.00   sec  59.6 MBytes   500 Mbits/sec  43156
[  5]   9.00-10.00  sec  59.6 MBytes   500 Mbits/sec  43152
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5]   0.00-10.00  sec   596 MBytes   500 Mbits/sec  0.000 ms  0/431611 (0%)  sender
[  5]   0.00-10.04  sec   558 MBytes   466 Mbits/sec  0.007 ms  27766/431591 (6.4%)  receiver

iperf Done.
@ubuntu:~$ iperf3  -c 169.254.254.188 -u -b 700M -t 10
Connecting to host 169.254.254.188, port 5201
[  5] local 169.254.254.180 port 54562 connected to 169.254.254.188 port 5201
[ ID] Interval           Transfer     Bitrate         Total Datagrams
[  5]   0.00-1.00   sec  83.4 MBytes   700 Mbits/sec  60417
[  5]   1.00-2.00   sec  83.4 MBytes   700 Mbits/sec  60427
[  5]   2.00-3.00   sec  83.4 MBytes   700 Mbits/sec  60415
[  5]   3.00-4.00   sec  83.4 MBytes   700 Mbits/sec  60427
[  5]   4.00-5.00   sec  83.4 MBytes   700 Mbits/sec  60420
[  5]   5.00-6.00   sec  83.5 MBytes   700 Mbits/sec  60458
[  5]   6.00-7.00   sec  83.5 MBytes   700 Mbits/sec  60432
[  5]   7.00-8.00   sec  83.4 MBytes   700 Mbits/sec  60420
[  5]   8.00-9.00   sec  83.4 MBytes   700 Mbits/sec  60389
[  5]   9.00-10.00  sec  83.5 MBytes   700 Mbits/sec  60458
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5]   0.00-10.00  sec   834 MBytes   700 Mbits/sec  0.000 ms  0/604263 (0%)  sender
[  5]   0.00-10.04  sec   569 MBytes   476 Mbits/sec  0.017 ms  191826/604219 (32%)  receiver

iperf Done.

**Comment by davidBar-On** [comment]
Hi,
Thanks for the input from the tests you run it is quite clear that iperf3 performance on Windows is limited and that at least for high bandwidth tests iperf2 is preferred in such environments.

Since formally iperf3 is not supported on Windows, I am not sure if this issue will be handled, especially if the problem is because of using Cygwin.

**Comment by xia-s-xian** [comment]
Hi:

Got it. Recently, I browsed the discussion board and saw that many people have encountered the same problem. If possible, I hope someone can fix this problem.

Thank you.
